llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -68,20 +71,26 @@ using namespace llvm;
 static cl::opt<bool> EnableRecPhiAnalysis("basic-aa-recphi", cl::Hidden,
                                           cl::init(true));
 
+/// Enable full analysis of PHI nodes.
+static cl::opt<bool> EnableFullPHIAnalysis("basic-aa-full-phi-analysis",
+                                           cl::Hidden, cl::init(false));
+
```

Comment:
I am worried that EnableFullPhiAnalysis=true relies on EnableRecPhiAnalysis=true for correctness.
For instance, there is a conservative return at line 1450 that is controlled by isRecursive, which is controlled by EnableRecPhiAnalysis.

Reply:
Discussed offline: if we use the && of these two conditions it looks mych safer.

Reply:
Done!